R/misc_yhat.R

```diff
@@ -149,7 +154,12 @@ yhat.ranger <- function(X.model, newdata, ...) {
     # if result is a vector then ncol parameter is null
     if (is.null(ncol(pred))) return(pred)
     # binary classification
-    if (ncol(pred) == 2) return(pred[,2])
+    if (ncol(pred) == 2 & !is.null(attr(X.model, "predict_function_column"))) {
+      pred <- pred[,attr(X.model, "predict_function_column")]
+    }
```

Comment:
try to avoid `}` in one line and `else` in another as this leads to errors that are hard to debug 
`} else` will work